hedera-node/hedera-addressbook-service-impl/src/main/java/com/hedera/node/app/service/addressbook/impl/validators/AddressBookValidator.java

```diff
@@ -0,0 +1,124 @@
+     */
+    public void validateDescription(@Nullable final String description, @NonNull final NodesConfig nodesConfig) {
+        if (description == null || description.isEmpty()) {
+            return;
+        }
+        final var raw = description.getBytes(StandardCharsets.UTF_8);
+        final var maxUtf8Bytes = nodesConfig.nodeMaxDescriptionUtf8Bytes();
+        validateFalse(raw.length > maxUtf8Bytes, INVALID_NODE_DESCRIPTION);
+        validateFalse(containsZeroByte(raw), INVALID_NODE_DESCRIPTION);
+    }
+
+    private boolean containsZeroByte(@NonNull final byte[] bytes) {
+        boolean ret = false;
+        for (final byte b : bytes) {
+            if (b == 0) {
+                ret = true;
+                break;
+            }
+        }
+        return ret;
+    }
+
+    /**
+     * Validates the gossip endpoint.
+     *
+     * @param endpointList The list of GossipEndpoint to validate
+     * @param nodesConfig The nodes configuration
+     */
+    public void validateGossipEndpoint(
+            @Nullable final List<ServiceEndpoint> endpointList, @NonNull final NodesConfig nodesConfig) {
+        validateFalse(endpointList == null || endpointList.isEmpty(), INVALID_GOSSIP_ENDPOINT);
+        validateFalse(endpointList.size() > nodesConfig.maxGossipEndpoint(), GOSSIP_ENDPOINTS_EXCEEDED_LIMIT);
+        // for phase 2: The first in the list is used as the Internal IP address in config.txt,
+        // the second in the list is used as the External IP address in config.txt
+        validateFalse(endpointList.size() < 2, INVALID_GOSSIP_ENDPOINT);
+
+        for (final var endpoint : endpointList) {
+            validateFalse(
+                    nodesConfig.gossipFqdnRestricted() && !endpoint.domainName().isEmpty(),
+                    GOSSIP_ENDPOINT_CANNOT_HAVE_FQDN);
+            validateEndpoint(endpoint, nodesConfig);
+        }
+    }
+
+    /**
+     * Validates the service endpoint.
+     *
+     * @param endpointList The list of ServiceEndpoint to validate
+     * @param nodesConfig The nodes configuration
+     */
+    public void validateServiceEndpoint(
+            @Nullable final List<ServiceEndpoint> endpointList, @NonNull final NodesConfig nodesConfig) {
+        validateFalse(endpointList == null || endpointList.isEmpty(), INVALID_SERVICE_ENDPOINT);
+        validateFalse(endpointList.size() > nodesConfig.maxServiceEndpoint(), INVALID_SERVICE_ENDPOINT);
+        for (final var endpoint : endpointList) {
+            validateEndpoint(endpoint, nodesConfig);
+        }
+    }
+
+    private void validateEndpoint(@NonNull final ServiceEndpoint endpoint, @NonNull final NodesConfig nodesConfig) {
+        validateFalse(endpoint.port() == 0, INVALID_ENDPOINT);
+        validateFalse(
+                endpoint.ipAddressV4().length() == 0
+                        && endpoint.domainName().trim().isEmpty(),
+                INVALID_ENDPOINT);
+        validateFalse(
+                endpoint.ipAddressV4().length() != 0
```

Comment:
All we should use is to check the ipAddressV4 is not empty? Or should we check if it follows some regex?

Reply:
I will add regex check in the next PR. Thanks @Neeharika-Sompalli